src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```
@@ -60,6 +61,24 @@ public function load(mixed $resource, ?string $type = null): mixed
         // Force load ContainerConfigurator to make env(), param() etc available.
         class_exists(ContainerConfigurator::class);
 
+        $autoloaderRegistered = false;
+        if (null === self::$configBuilderAutoloader) {
+            $generator = $this->generator;
+            self::$configBuilderAutoloader = function (string $class) use ($loader, $generator) {
+                if (!$generator || !str_starts_with($class, 'Symfony\\Config\\') || !str_ends_with($class, 'Config')) {
+                    return;
+                }
+
+                try {
+                    $loader->configBuilder($class);
+                } catch (\LogicException|\InvalidArgumentException) {
+                    // no-op
```

Do you know which error can be hidden here? Hiding generic exceptions like this can make debugging complicated.

Note that `InvalidArgumentException` extends `LogicException`, so it can be removed.

---

Right. I removed the try/catch block to propagate the exception instead of waiting and having an obscure "Class not found" later.